### DISCUSSION TITLE
Query cancellation with React Query and graphql-codegen

### DISCUSSION THREAD

**Discussion by mabecth** [TanStack/query#4551] [Q&A]
I am using React Query for my graphql code and I also generate types from graphql-codegen. I want my queries to auto cancel on onmount etc when passing a signal to my fetcher like the docs mention. The problem is, how do I pass the signal instance to the fetcher when the fetcher is only called in the graphql-codegen generated code? Looking at the docs for the React Query graphql-codegen plugin there seems to be suitable option. I guess I could create a signal instance manually but then it would not hook into React Query and automatically cancel queries.

**Comment by TkDodo** [comment]
I think this is a question you have to ask in the `graphql-codegen` repo, as they are generating the code ?
